lib/sycamore/sycamore/connectors/pinecone/pinecone_reader.py

```diff
@@ -0,0 +1,77 @@
+from sycamore.data import Document
+from pinecone.grpc import PineconeGRPC
+from sycamore.connectors.common import unflatten_data
+from sycamore.connectors.base_reader import BaseDBReader
+from dataclasses import dataclass
+from typing import Optional, Dict
+
+
+@dataclass
+class PineconeReaderClientParams(BaseDBReader.ClientParams):
+    api_key: str
+
+
+@dataclass
+class PineconeReaderQueryParams(BaseDBReader.QueryParams):
+    index_name: str
+    namespace: str
```

Comment:
maybe `namespace: str = ""`

Reply:
Already set in the reader entry point, don't want to repeat it